File: mirea/cources/2026-2027/aii-july/01/Part-03-Sorting/examples/example-09.py
# Сортировка списка с key-функцией в Python3

# Чтение списка
n = int(input())
a = list(map(int, input().split()))

# В Python3 обычно задают key, а не собственный компаратор
a.sort(key=lambda x: (abs(x), x))

# Вывод результата
print(*a)

File: mirea/cources/2026-2027/aii-july/01/Part-03-Sorting/examples/example-09.cpp
// Сортировка массива с компаратором в C++

#include <algorithm>
#include <cmath>
#include <iostream>
#include <vector>

using namespace std;

int main() {
    // Чтение массива
    int n;
    cin >> n;

    vector<int> a(n);
    for (int i = 0; i < n; i++) {
        cin >> a[i];
    }

    // Сортируем по возрастанию абсолютного значения,
    // а при равенстве модулей — по самому числу
    sort(a.begin(), a.end(), [](int x, int y) {
        if (abs(x) != abs(y)) {
            return abs(x) < abs(y);
        }
        return x < y;
    });

    // Вывод результата
    for (int x : a) {
        cout << x << ' ';
    }
    cout << '\n';
}